mget.go

```diff
@@ -95,3 +96,27 @@ type MGetResult struct {
 func (m MGetResult) GetSource() []byte {
 	return []byte(m.Source)
 }
+
+// ValidateForVersion checks for validation in the [MGetRequest] instance relating to the version of OpenSearch given by v.
+func (m MGetRequest) ValidateForVersion(v Version) (ValidationResults, error) {
+	validationResults := make(ValidationResults, 0)
+
+	switch v {
+	case V2:
```

Comment:
Agreed.  I would export `osv2.ValidateMGet() (ValidationResults, error)` or something similar and have this case delegate to that.

Reply:
Ya, I agree. I started it that way but it's going to require some re-jigging because of package cycles...

Reply:
Hmm, that might indicate we're fighting against an anti-pattern of some sort...

Reply:
On the other hand, why should the osv2 package need to have knowledge about the weird version-agnostic union types defined in the root? I'm starting to wonder if maybe the reason this cycle is proving hard to break is because this logic validation actually does belong here in the root / outside of osv2.

Reply:
Yeah.  I wasn't thinking about the import cycle.  The root is the "domain" so it shouldn't be importing sub-packages.

I think I would still refactor the code into a separate unexported function to unclutter this method.  Maybe create `validation_osv[N].go` and have the validation logic for each version be in its own file?

Reply:
Ya, that makes sense.
I also think that I should move the conversion related methods and function in osv2/mget.go out of that package. Since again, the stuff in osv2 should be standalone and not be cluttered up with the version agnostic stuff.

Reply:
👍 as long as `osv2` is importing the root and not the other way around.